rest/jetty/src/main/scala/io/udash/rest/jetty/JettyRestClient.scala

```diff
@@ -3,86 +3,305 @@ package rest.jetty
+                val mediaTypeOpt = contentTypeOpt.map(MimeTypes.getContentTypeWithoutCharset)
+                val charsetOpt = contentTypeOpt.map(MimeTypes.getCharsetFromContentType)
+                val bodyOpt = (mediaTypeOpt, charsetOpt) matchOpt {
+                  case (Opt(HttpBody.JsonType), Opt(charset)) =>
+                    // suboptimal - maybe "online" parsing is possible using Jackson / other lib without waiting for full content ?
+                    StreamedBody.JsonList(
+                      elements = Observable
+                        .fromTask(Utils.mergeArrays(rawContentSubject))
+                        .map(raw => new String(raw, charset))
+                        .flatMap { jsonStr =>
+                          val input = new JsonStringInput(new JsonReader(jsonStr))
+                          Observable
+                            .fromIterator(Task.eval(input.readList().iterator(_.asInstanceOf[JsonStringInput].readRawJson())))
+                            .map(JsonValue(_))
+                        }
+                        .doOnSubscriptionCancel(cancelRequest)
+                        .onErrorFallbackTo(Observable.raiseError(JettyRestClient.Streaming)),
+                      charset = charset,
+                    )
+                  case (Opt(mediaType), _) =>
+                    StreamedBody.binary(
+                      content = rawContentSubject.doOnSubscriptionCancel(cancelRequest),
+                      contentType = contentTypeOpt.getOrElse(mediaType),
+                    )
+                }
+                bodyOpt.mapOr(
+                  {
+                    callback(Failure(JettyRestClient.unsupportedContentTypeError(contentTypeOpt)))
+                  },
+                  body => {
+                    this.collectToBuffer = false
+                    val restResponse = StreamedRestResponse(
+                      code = response.getStatus,
+                      headers = parseHeaders(response),
+                      body = body,
+                    )
+                    callback(Success(restResponse))
+                  }
+                )
+              }
+            }
+
+            override def onContent(response: Response, chunk: Content.Chunk, demander: Runnable): Unit =
+              if (collectToBuffer)
+                super.onContent(response, chunk, demander)
+              else if (chunk == Content.Chunk.EOF) {
+                rawContentSubject.onComplete()
+              } else {
+                val buf = chunk.getByteBuffer
+                val arr = new Array[Byte](buf.remaining)
+                buf.get(arr)
+                publishSubject.subscription // wait for subscription
+                  .flatMapNow(_ => rawContentSubject.onNext(arr))
+                  .mapNow {
+                    case Ack.Continue => demander.run()
+                    case Ack.Stop     => ()
+                  }
```

Comment:
that's a discarded Future - if it's intended, it's not clear to me whether it's intended

Reply:
@sebaciv any idea?